### PR TITLE
Review of Milestone 2.

### PR DESCRIPTION
**Pull request by cod3smith**
This PR herein contains the following:
1. A way for a user to add friends.
2. Unit tests for all models.
3. Integration tests for the user view.

### CODE REVIEW

**Review by trekab — CHANGES_REQUESTED** [review]
### Changes Required! :recycle: 
Hi team! Well done implementing the required relationships amongst your models. There are some requirements that you still need to implement to complete this milestone though. These include:
- [x] A user should be able to see friendship requests ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/pending_request.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- [x] A user has to accept friendship request to start friendship ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- [x] A user should be able to reject a friendship request ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- [x] The button to invite a user to friendship should appear on both the users' list page and on a single user's page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- [x] Posts timeline should include posts written by the logged-in user and his/her friends. Your current version displays all posts to any logged-in user. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- [x] Well done writing the tests for your associations. Consider adding some validation tests and integration tests. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/tests.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- [x] __[Note]__ It appears that you set up your linting correctly but it is still unable to run the checks on your pull request. Consider replacing your config files with new ones and see if this persists.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__
[GitHub](https://github.com/trekab) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

**Comment by abouhid** [comment]
All of the issues were fixed and we even took it further and finished the following two milestones! Thank you for the help.

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
+    <article class="timeline">
+     
+    <% if @user.friendship_requested?(current_user) %>
+
+    <ul class="posts">
+      <%= render "friend_request" %>
+    </ul>
+    <% end %> 
+
+ <% unless current_user.friendship_requested?(@user) || current_user.friend?(@user) %>
```
Comment (kubilaycaglayan) [inline app/views/users/show.html.erb]: :bulb: There should be no logic in the views.

**Review by kubilaycaglayan — CHANGES_REQUESTED** [review]
Hello Team 👋
Status: Changes Required ♻️

Great job! You fulfilled the requirements that the previous TSE mentioned. 👍👏😊
There's just one thing I noticed when looking at show.html.erb in users view.
We should keep the logic away from the views. 
Please work on it, I am sharing the necessary documentation with you.


### - [ ] [REQUIRED] There shoud be absolutely NO logic in views.

>Views are only for displaying information that became available to it from its corresponding controller not generating it.

**Resources** 
- [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
- [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
- [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

### That's it, team! After making the changes you can request another code review 😇

With all my respect 🙏
> Kubilay Caglayan | Technical Support Engineer

**Review by kobiyoyo — APPROVED** [review]
# Project Approved :thumbsup:
## Hello Team  ,  you have done a brilliant job with this project. :clap:
#### You can merge !!!

**Review by voscarmv — APPROVED** [review]
# Project status: Approved ✨ 😎👍 💯 ✔️
### You may merge this branch and move forward ⛙🛣

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Pull request merged**